python/hail/methods/statgen.py

```diff
@@ -1893,6 +1899,140 @@ def row_correlation(entry_expr, block_size=None) -> BlockMatrix:
+def ld_matrix(entry_expr, locus_expr, radius, coord_expr=None, block_size=None) -> BlockMatrix:
+    """Computes the windowed correlation (linkage disequilibrium) matrix between
+    variants.
+
+    Examples
+    --------
+    Consider the following dataset consisting of three variants with centimorgan
+    coordinates and four samples:
+
+    >>> data = [{'v': '1:1:A:C',       'cm': 0.1, 's': 'a', 'GT': hl.Call([0, 0])},
+    ...         {'v': '1:1:A:C',       'cm': 0.1, 's': 'b', 'GT': hl.Call([0, 0])},
+    ...         {'v': '1:1:A:C',       'cm': 0.1, 's': 'c', 'GT': hl.Call([0, 1])},
+    ...         {'v': '1:1:A:C',       'cm': 0.1, 's': 'd', 'GT': hl.Call([1, 1])},
+    ...         {'v': '1:2000000:G:T', 'cm': 0.9, 's': 'a', 'GT': hl.Call([0, 1])},
+    ...         {'v': '1:2000000:G:T', 'cm': 0.9, 's': 'b', 'GT': hl.Call([1, 1])},
+    ...         {'v': '1:2000000:G:T', 'cm': 0.9, 's': 'c', 'GT': hl.Call([0, 1])},
+    ...         {'v': '1:2000000:G:T', 'cm': 0.9, 's': 'd', 'GT': hl.Call([0, 0])},
+    ...         {'v': '2:1:C:G',       'cm': 0.2, 's': 'a', 'GT': hl.Call([0, 1])},
+    ...         {'v': '2:1:C:G',       'cm': 0.2, 's': 'b', 'GT': hl.Call([0, 0])},
+    ...         {'v': '2:1:C:G',       'cm': 0.2, 's': 'c', 'GT': hl.Call([1, 1])},
+    ...         {'v': '2:1:C:G',       'cm': 0.2, 's': 'd', 'GT': hl.null(hl.tcall)}]
+    >>> ht = hl.Table.parallelize(data, hl.dtype('struct{v: str, s: str, cm: float64, GT: call}'))
+    >>> ht = ht.transmute(**hl.parse_variant(ht.v))
+    >>> mt = ht.to_matrix_table(row_key=['locus', 'alleles'], col_key=['s'], row_fields=['cm'])
+
+    Compute linkage disequilibrium between all pairs of variants on the same
+    contig and within one megabase:
+
+    >>> ld = hl.ld_matrix(mt.GT.n_alt_alleles(), mt.locus, radius=1e6)
+    >>> ld.to_numpy()
+    array([[1., 0., 0.],
+           [0., 1., 0.],
+           [0., 0., 1.]])
+
+    Within two megabases:
+
+    >>> ld = hl.ld_matrix(mt.GT.n_alt_alleles(), mt.locus, radius=2e6)
+    >>> ld.to_numpy()
+    array([[ 1.        , -0.85280287,  0.        ],
+           [-0.85280287,  1.        ,  0.        ],
+           [ 0.        ,  0.        ,  1.        ]])
+
+    Within one centimorgan:
+
+    >>> ld = hl.ld_matrix(mt.GT.n_alt_alleles(), mt.locus, radius=1.0, coord_expr=mt.cm)
+    >>> ld.to_numpy()
+    array([[ 1.        , -0.85280287,  0.        ],
+           [-0.85280287,  1.        ,  0.        ],
+           [ 0.        ,  0.        ,  1.        ]])
+
+    Within one centimorgan, and only calculate the upper triangle:
+
+    >>> ld = hl.ld_matrix(mt.GT.n_alt_alleles(), mt.locus, radius=1.0, coord_expr=mt.cm)
+    >>> ld = ld.sparsify_triangle()
+    >>> ld.to_numpy()
+    array([[ 1.        , -0.85280287,  0.        ],
+           [ 0.        ,  1.        ,  0.        ],
+           [ 0.        ,  0.        ,  1.        ]])
+
+    Notes
+    -----
+    This method applies :meth:`row_correlation` and sparsifies the result.
```

Comment:
suggestion: be more precise what `sparsifies` means in this context.

Reply:
good catch